### DISCUSSION TITLE
Picture size in wfb-ng-osd on Debian12

### DISCUSSION THREAD

**Discussion by avs76** [svpcom/wfb-ng#393] [Questions]
Hi!
I builds osd app on desktop Debian12 x64 for tests.
When opened RTSP stream, image size sets to 320x240.

![image](https://github.com/user-attachments/assets/ffaa7c5a-d7a4-4513-bce0-99f9881c3f40)

If I set parametr -w 100, image changes to low resolution, size also 320x240.

![image](https://github.com/user-attachments/assets/0aa8904d-1370-4d9b-a80f-b0f67b2a6a36)

When I use gst-launch the image looks good in full HD.
How can I change the size of the image displayed on the screen?
Thanks.

**Comment by svpcom** [comment]
@avs76 What video card / window manager do you use?  Try to comment-out ` setenv("GST_GL_PLATFORM", "egl", 0);` in appsrc.c

**Reply by avs76** [reply]
Video: Intel® UHD Graphics 630

Window manager:

```
Version
=======
KWin version: 5.27.5
Qt Version: 5.15.8
Qt compile version: 5.15.8
XCB compile version: 1.15

Operation Mode: Xwayland

Build Options
=============
KWIN_BUILD_DECORATIONS: yes
KWIN_BUILD_TABBOX: yes
KWIN_BUILD_ACTIVITIES: yes
HAVE_X11_XCB: yes
HAVE_EPOXY_GLX: yes

X11
===
Vendor: The X.Org Foundation
Vendor Release: 12201009
Protocol Version/Revision: 11/0
SHAPE: yes; Version: 0x11
RANDR: yes; Version: 0x14
DAMAGE: yes; Version: 0x11
Composite: yes; Version: 0x4
RENDER: yes; Version: 0xb
XFIXES: yes; Version: 0x50
SYNC: yes; Version: 0x31
GLX: yes; Version: 0x0
```

**Reply by svpcom** [reply]
It is very strange. wfb-ng-osd window should have standard window decorations and control and allow to be resizable

**Reply by avs76** [reply]
Yes, and commenting setenv("GST_GL_PLATFORM", "egl", 0); in appsrc.c, had no effect.
And as you can see in the screenshots, the window has no controls, is located in the center of the screen and has a size of 320x240.

**Reply by svpcom** [reply]
Try to run `GST_GL_PLATFORM=egl  gst-launch-1.0 gltestsrc ! glimagesink`  it should produce resizeable window.

**Reply by avs76** [reply]
Window 320x240 no resizable and without controls.

![image](https://github.com/user-attachments/assets/10f06cad-d6e0-4207-8ed4-31d32b5ec24b)

**Reply by avs76** [reply]
Ok, thanks.
Maybe I need to change window manager.
I'll try to build on Raspberry 4.

**Reply by svpcom** [reply]
It seems that is something bad with your window manager. Try other (for example gnome or xfce4)
